### DISCUSSION TITLE
Provide a string.Equals(StringComparison.OrdinalIgnoreCase) equivalent operator

### DISCUSSION THREAD

**Discussion by ahdung** [dotnet/csharplang#4117] [Language Ideas]
I hope this code:
```c#
string.Equals(a, b, StringComparison.OrdinalIgnoreCase)
```
can replaced by:
```c#
a ≈ b
```
may be other like: `~~`, `=~`, `=:=`

**Comment by YairHalberstadt** [comment]
Why is this case so important that it requires a special operator? `StringComparison` has 6 members. Adding a new operator for all of them is clearly not practical, so what makes `OrdinalIgnoreCase` the one that merits special treatment?

**Comment by huoyaoyuan** [comment]
You can write an extension method for this to save typing.

**Comment by orthoxerox** [comment]
Or a wrapper `CaseInsensitiveString` class around `string` that implements `==`.

**Comment by theunrepentantgeek** [comment]
In most (in fact, the vast majority of) cases where I initially end up writing code like this:
```
string.Equals(myCustomerId, otherCustomerId, StringComparison.OrdinalIgnoreCase)
```
it's because there's a semantic type that needs to be explicitly identified - in this example `CustomerId`.

Pull that out as a struct (as it's a value type) and you can override `==` to give you the case insensitivity you desire:

```
myCustomerId == otherCustomerId
```

Moreover, you gain certainty that everywhere you compare two CustomerIds, you are doing so the same way. 

(Not to mention the type safety of not confusing a `CustomerId` with a `ProductId`, the ability to customize sorting, a natural location place to implement parsing, and the ability to enforce invariants for validation).
